src/components/authoring.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from "react";
+import { BaseComponent, IBaseProps } from "./base";
+
+import "./app.sass";
+import Form, { ISubmitEvent } from "react-jsonschema-form";
+import { AppMode } from "../models/stores";
+import { QueryParams } from "../utilities/url-params";
+
+interface IProps extends IBaseProps {}
+interface IState {}
+
+export class AuthoringComponent extends BaseComponent<IProps, IState> {
+  public render() {
+    const onSubmit = (e: ISubmitEvent<QueryParams>) => {
+      const encodedParams = encodeURIComponent(JSON.stringify(e.formData));
+      window.open(`${location.origin}${location.pathname}?${encodedParams}`, "connected-bio-spaces");
+    };
+    return (
+      <div className="authoring">
+        <Form
+          schema={{
+            title: "Connected Bio Parameters",
+            type: "object",
+            properties: {
+              topBar: {type: "boolean", title: "Show Top Bar?", default: true},
```

Comment:
Is hiding the top bar part of a larger embedded mode?  If so, is it better to have a singular embedded mode or let the user mix and match features with the authoring tool?   Either way, this seems fine for now.

Reply:
Oh good question...agreed I'll leave it for now, but if more features depend on this it would be worth changing the URL parameter

Reply:
Agreed.  An embedded mode might be nothing more than hiding the topbar.  I did briefly work on some changes in CODAP's embedded mode (for when it gets placed in LARA or other apps) recently, and it has a variety of minor, nuanced things that it does when embedded.  Regardless, probably a question for another day.

Reply:
This is another reason to hoist URL processing to index.tsx. The AppComponent render function shouldn't have to change if we decide to change the URL parameter to a mode. Having `showTopBar` be a prop puts that responsibility where it makes more sense.